src/frontend/src/instance/distributed.rs

```diff
@@ -266,6 +266,62 @@ impl DistInstance {
         Ok(Output::AffectedRows(1))
     }
 
+    async fn flush_table(&self, table_name: TableName, region_id: Option<u32>) -> Result<Output> {
+        let _ = self
+            .catalog_manager
+            .table(
+                &table_name.catalog_name,
+                &table_name.schema_name,
+                &table_name.table_name,
+            )
+            .await
+            .context(CatalogSnafu)?
+            .with_context(|| TableNotFoundSnafu {
+                table_name: table_name.to_string(),
+            })?;
+
+        let route_response = self
+            .meta_client
+            .route(RouteRequest {
+                table_names: vec![table_name.clone()],
+            })
+            .await
+            .context(RequestMetaSnafu)?;
+
+        let expr = FlushTableExpr {
+            catalog_name: table_name.catalog_name.clone(),
+            schema_name: table_name.schema_name.clone(),
+            table_name: table_name.table_name.clone(),
+            region_id,
+        };
+
+        for table_route in route_response.table_routes.iter() {
+            let should_send_rpc = table_route.region_routes.iter().any(|route| {
+                if let Some(region_id) = region_id {
+                    // ignores the table id
+                    u64::from(region_id) == route.region.id & u64::from(u32::MAX)
+                } else {
+                    true
+                }
```

Comment:
Looks like the `region_id` is actually a `region_number`.

@v0y4g3r We might need to provide a common method to encode/decode the region id. We should not expose the detail of region id to the frontend. Do you have any better idea to filter the region?

Reply:
Maybe implement `From<RegionId>` for `RegionNumber`?

Reply:
> Looks like the `region_id` is actually a `region_number`.
> 
> @v0y4g3r We might need to provide a common method to encode/decode the region id. We should not expose the detail of region id to the frontend. Do you have any better idea to filter the region?

Maybe implement `From<RegionId>` for `RegionNumber`?

Reply:
The `route.region.id` is a `u64` datatype, but here the `region_id` is a `u32` datatype, then I guess it may be defined as the following: 
```
#[inline]
pub fn region_id(table_id: TableId, n: u32) -> RegionId {
    (u64::from(table_id) << 32) | u64::from(n)
}
```

But, I step into the tests, and the value of `route.region.id` starts from `1` (not the `1024 << 32 | 0` ). It seems like the `route.region.id` is a `u64` version of `region_id.`

Reply:
> The `route.region.id` is a `u64` datatype, but here the `region_id` is a `u32` datatype, then I guess it may be defined as the following:
> 
> ```
> #[inline]
> pub fn region_id(table_id: TableId, n: u32) -> RegionId {
>     (u64::from(table_id) << 32) | u64::from(n)
> }
> ```
> 
> But, I step into the tests, and the value of `route.region.id` starts from `1` (not the `1024 << 32 | 0` ). It seems like the `route.region.id` is a `u64` version of `region_id` (or something called `RegionNumber`)

I believe it's a legacy problem. `route.region.id` is actually `RegionNumber`, which starts from 0 to (region number-1)

https://github.com/GreptimeTeam/greptimedb/blob/803b7f0633b731b1742863b530b97b1ff2e1a07a/src/meta-srv/src/service/router.rs#L133-L138

`RegionId` contains both `TableId` in the most signifcant 32 bits and `RegionNumber` in the least significant 32 bits.

https://github.com/GreptimeTeam/greptimedb/blob/e17d5a1c41147fd8033cf87017c5099a53158d34/src/mito/src/engine.rs#L65-L68

Reply:
This is quite confusing 😢